python/lsst/ip/diffim/subtractImages.py

```diff
@@ -273,12 +274,34 @@ def run(self, template, science, sources, finalizedPsfApCorrCatalog=None):
                                             finalizedPsfApCorrCatalog=finalizedPsfApCorrCatalog)
         checkTemplateIsSufficient(template, self.log,
                                   requiredTemplateFraction=self.config.requiredTemplateFraction)
-        sciencePsfSize = getPsfFwhm(science.psf)
-        templatePsfSize = getPsfFwhm(template.psf)
+
+        # In the event that getPsfFwhm fails, evaluate the PSF on a grid.
+        fwhmExposureBuffer = self.config.makeKernel.fwhmExposureBuffer
+        fwhmExposureGrid = self.config.makeKernel.fwhmExposureGrid
+
+        try:
+            templatePsfSize = getPsfFwhm(template.psf)
+            sciencePsfSize = getPsfFwhm(science.psf)
+        except InvalidParameterError:
+            self.log.info("Unable to evaluate PSF at the average position. "
+                          "Evaluting PSF on a grid of points."
+                          )
+            templatePsfSize = evaluateMeanPsfFwhm(template,
+                                                  fwhmExposureBuffer=fwhmExposureBuffer,
+                                                  fwhmExposureGrid=fwhmExposureGrid
+                                                  )
+            sciencePsfSize = evaluateMeanPsfFwhm(science,
+                                                 fwhmExposureBuffer=fwhmExposureBuffer,
+                                                 fwhmExposureGrid=fwhmExposureGrid
+                                                 )
```

Comment:
I think this should be split into two try/except blocks, because it's more likely that only one should fail.  Unless there's a reason that if one fails they both should be recomputed that I'm missing?

Reply:
Also, what happens if the `evaluateMeanPsfFwhm` fails? We get a `ValueError` and is that caught anywhere?

Reply:
The expectation, based on what Ian had said, is that only template could have holes and not the science images. The error specifically comes from `CoaddPsf` and I believe the PSF of science images are _not_ `CoaddPsf`s. So the failures come from template, which is what we try to do first.

They are not in two try/except blocks because we'd want to have the same definition of PSF size for both the images. Having two different blocks would mean their sizes can get defined differently.

Reply:
My understanding is if we get a `ValueError`, it should indeed crash the pipeline (?). This would only happen if our grid happens to exactly line up with chip gaps or dithering strategy.

Reply:
Can you make this comment in the code for why these are in the same try/except block?

Reply:
And it is done!